include/libdnf/advisory/advisory_package.hpp

```diff
@@ -0,0 +1,104 @@
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_ADVISORY_ADVISORY_PACKAGE_HPP
+#define LIBDNF_ADVISORY_ADVISORY_PACKAGE_HPP
+
+#include "libdnf/advisory/advisory.hpp"
+
+#include <memory>
+#include <vector>
+
+
+namespace libdnf::advisory {
+
+class AdvisoryPackage {
+public:
+    AdvisoryPackage(const AdvisoryPackage & src);
+    AdvisoryPackage(AdvisoryPackage && src);
+
+    AdvisoryPackage & operator=(const AdvisoryPackage & src);
+    AdvisoryPackage & operator=(AdvisoryPackage && src) noexcept;
+    ~AdvisoryPackage();
+
+    /// Get name of this AdvisoryPackage.
+    ///
+    /// @return Name of this AdvisoryPackage as std::string.
+    std::string get_name() const;
+
+    /// Get version of this AdvisoryPackage.
+    ///
+    /// @return Version of this AdvisoryPackage as std::string.
+    std::string get_version() const;
+
+    /// Get evr of this AdvisoryPackage.
+    ///
+    /// @return Evr of this AdvisoryPackage as std::string.
+    std::string get_evr() const;
+
+    /// Get arch of this AdvisoryPackage.
+    ///
+    /// @return Arch of this AdvisoryPackage as std::string.
+    std::string get_arch() const;
+
+    /// Get AdvisoryId of Advisory this AdvisoryPackage belongs to.
+    ///
+    /// @return AdvisoryId of this AdvisoryPackage.
+    AdvisoryId get_advisory_id() const;
+
+    //TODO(amatej): we should be able to get Advisory and AdvisoryCollection information
+    //              from a package, returning new advisory object is one option but it might
+    //              be better to set up so sort of inheritance among Advisory, AdvisoryCollection,
+    //              AdvisoryPacakge and AdvisoryModule.
+    /// Get Advisory this AdvisoryPackage belongs to.
+    ///
+    /// @return newly construted Advisory object of this AdvisoryPackage.
+    Advisory get_advisory() const;
+    /// Get AdvisoryCollection this AdvisoryPackage belongs to.
+    ///
+    /// @return newly construted AdvisoryCollection object of this AdvisoryPackage.
+    AdvisoryCollection get_advisory_collection() const;
+
+    /// Compare NEVRAs of two AdvisoryPackages
+    ///
+    /// @param first        First AdvisoryPackage to compare.
+    /// @param second       Secondf AdvisoryPackage to compare.
+    /// @return True if first AdvisoryPackage has smaller NEVRA, False otherwise.
+    static bool compareNEVRA(const AdvisoryPackage & first, const AdvisoryPackage & second);
+
+    /// Compare Name and Architecture of AdvisoryPackage and libdnf::rpm::Package
+    ///
+    /// @param adv_pkg      AdvisoryPackage to compare.
+    /// @param pkg          libdnf::rpm::Package to compare.
+    /// @return True if AdvisoryPackage has smaller name or architecture than libdnf::rpm::package, False otherwise.
+    static bool comparePackageNA(const AdvisoryPackage & adv_pkg, const rpm::Package & pkg);
```

Comment:
I am not sure about name of the method. It compares two advisories but it answers question - is lower.

Reply:
I have moved all the comparing methods into `Impl` of `AdvisoryPackage` and renamed them to better match the naming scheme set up in `SolvQuery` but just to be sure I have also added a TODO to check if the names are appropriate.